doc/compiler/jitserver/ClientSession.md

```diff
@@ -27,19 +27,26 @@ Some of the most commonly used items in the client session data are the cached R
 
 ### Adding new items to the client session
 1. Modify the relevant structure (`ClientSessionData`, `ClassInfo`, or `J9MethodInfo`) to contain your data. See below for information about what goes where.
-1. Modify ClassInfoTuple (add to end)
-1. Modify cacheRemoteROMClass packRemoteROMClassInfo
+2. Modify ClassInfoTuple (add to end)
+3. Modify cacheRemoteROMClass packRemoteROMClassInfo
 
 ### `ClientSessionData`
 Stores the data for an individual client session. During a compilation, this is accessible via the method `TR::CompilationInfoPerThread::getClientData()`, typically referenced as `_compInfoPT->getClientData()`. For access during other times and lifetime information, see [`ClientSessionHT`](#clientsessionht).
 
 This includes a few pointers to global objects in the JVM, the hash tables containing `ClassInfo` and `J9MethodInfo`, and the server copy of the CHTable.
 
 ### `ClassInfo`
-Stores cached data about a class loaded the client VM. This includes the ROM class, which is copied in full to the server, as well as other items which are just pointers to data on the client.
+Stores cached data about a class loaded on the client VM. This includes the ROM class, which is copied in full to the server, as well as other items which are just pointers to data on the client.`ClassInfo` will persist on the server until the corresponding Java class gets unloaded. At that point, server will be notified and the cache will be purged.
```

Comment:
"`ClassInfo` will persist on the server until the corresponding Java class gets unloaded or replaced by the HCR mechanism"

Reply:
Good point. But actually, server doesn't see the difference between HCR event and class unload. When class is redefined, jitserver will treat it the same as class being unloaded.